Rubberduck.Core/UI/CodeExplorer/Commands/AddComponentCommand.cs

```diff
@@ -51,6 +53,60 @@ public void AddComponent(CodeExplorerItemViewModel node, ComponentType type)
             }
         }
 
+        public void AddComponent(CodeExplorerItemViewModel node, string moduleText)
+        {
+            var nodeProject = GetDeclaration(node)?.Project;
+            if (node != null && nodeProject == null)
+            {
+                return; //The project is not available.
+            }
+
+            string optionCompare = string.Empty;
+            using (IHostApplication hostApp = _vbe.HostApplication())
+            {
+                optionCompare = hostApp.ApplicationName == "Access" ? "Option Compare Database" :
+                    string.Empty;
+            }
+
+            using (var components = node != null
+                ? nodeProject.VBComponents
+                : ComponentsCollectionFromActiveProject())
+            {
+                var folderAnnotation = $"'@Folder(\"{GetFolder(node)}\")";
+                string fileName = createTempTextFile(moduleText);
+
+                using (var newComponent = components.Import(fileName))
+                {
+                    using (var codeModule = newComponent.CodeModule)
+                    {
+                        var delarationLines = string.Concat(folderAnnotation,  optionCompare);
+                        if (optionCompare.Length > 0)
+                        {
+                            codeModule.InsertLines(1, optionCompare);
+                        }
+                        if (folderAnnotation.Length > 0)
+                        {
+                            codeModule.InsertLines(1, folderAnnotation);
+                        }
+                        codeModule.CodePane.Show();
+                    }
+                }
+                File.Delete(fileName);
+            }
+        }
+
+        private string createTempTextFile(string moduleText)
```

Comment:
I feel that this already has been coded somewhere else. I wonder if RD really needs to have a `TempFileService` here to avoid re-inventing the wheel?

Reply:
Something similar exists in the `TempSourceFileHandler`, but it exports an existing component.

Reply:
For now, this is probably best dealt in a separate PR. The `TempSourceFileHandler` does have a `ImportAndCleanUp` method but it is not implemented for VB6. Furthermore, it'd only work for an existing component because it deletes a component before importing. I will open an issue for that.